File: src_c/mat_con1.c
/**
 ** MAT_CON1
**/
#include <stdio.h>
#include "pfem_util.h"
#include "allocate.h"
extern FILE* fp_log;
void MAT_CON1()
{
  int i,k,kk;
  
  indexLU=(KINT*)allocate_vector(sizeof(KINT),NP+1);
  for(i=0;i<NP+1;i++) indexLU[i]=0;
  
  for(i=0;i<NP;i++){
    indexLU[i+1]=indexLU[i]+INLU[i];
  }
  
  NPLU=indexLU[NP];
  
  itemLU=(KINT*)allocate_vector(sizeof(KINT),NPLU);
  
  for(i=0;i<NP;i++){
    for(k=0;k<INLU[i];k++){
      kk=k+indexLU[i];
      itemLU[kk]=IALU[i][k]-1;
    }
  }
  
  deallocate_vector(INLU);
  deallocate_vector(IALU);
}

File: src_f/mat_con1.f
!C
!C***
!C*** MAT_CON1
!C***
!C
      subroutine MAT_CON1
      use pfem_util
      implicit REAL*8 (A-H,O-Z)

      allocate (index(0:NP))
      index= 0

      do i= 1, NP
        index(i)= index(i-1) + INLU(i)
      enddo

      NPLU= index(NP)

      allocate (item(NPLU))
      if (NFLAG.eq.0) then
        item= 0
       else
!$omp parallel do private(j,k)
        do j= 1, N
          do k= index(j-1)+1, index(j)
            item(k)= 0
          enddo
        enddo
      endif

      do i= 1, NP
        do k= 1, INLU(i)
               kk = k + index(i-1)
          item(kk)=      IALU(i,k)
        enddo
      enddo

      deallocate (INLU, IALU)

      end subroutine MAT_CON1
